In [13]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
import os
from sklearn.linear_model import Ridge
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error

In [14]:
os.chdir("D:/Machine Learning/Datasets/Datasets")

In [15]:
df = pd.read_csv('Housing.csv')

In [16]:
dum_df = pd.get_dummies(df,drop_first=True)

In [17]:
x = dum_df.drop(['price'],axis=1)
y = dum_df['price']

In [18]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.3,random_state=24)

In [40]:
ridge = Ridge(alpha=5)
ridge.fit(x_train, y_train)

Ridge(alpha=5)

In [41]:
y_pred = ridge.predict(x_test)

In [42]:
r2 = r2_score(y_test,y_pred)
print(f'r2 score: {r2}')

r2 score: 0.6221746783819945


In [43]:
print(ridge.coef_)

[3.73133298e+00 2.02856649e+03 1.35561840e+04 6.61438192e+03
 5.19230582e+03 6.69612836e+03 3.62510740e+03 7.08013520e+03
 8.12026342e+03 1.14889109e+04 7.16914747e+03]


In [44]:
MSE = mean_squared_error(y_test,y_pred)
MAE = mean_absolute_error(y_test,y_pred)
print("Mean squared error is : ", MSE)
print("Mean Absolute error is : ", MAE)

Mean squared error is :  264059424.4175551
Mean Absolute error is :  12235.665161309333


for 
alpha = 1, r2 score = 0.6243
alpha = 2, r2 score = 0.6237490936338193
alpha = 3, r2 score = 0.623240542878878
alpha = 5, r2 score = 0.6221746783819945

In [50]:
alphas = [0.5,1,1.5,2,2.5]
tmp =[]
for i in alphas:
    ridge = Ridge(alpha=i)
    ridge.fit(x_train, y_train)
    y_pred = ridge.predict(x_test)
    r2 = r2_score(y_test,y_pred)
    tmp.append(r2)
    print(f' for alpha={i}, r2 score: {r2}')

maxr2 = max(tmp)
print(maxr2 )

 for alpha=0.5, r2 score: 0.6244646032856002
 for alpha=1, r2 score: 0.6242338599592155
 for alpha=1.5, r2 score: 0.6239949379104519
 for alpha=2, r2 score: 0.6237490936338193
 for alpha=2.5, r2 score: 0.6234973500892027
0.6244646032856002


In [56]:
alphas = np.arange(0.25,10,0.25)
scores = []
for i in alphas:
    ridge = Ridge(alpha=i)
    ridge.fit(x_train, y_train)
    y_pred = ridge.predict(x_test)
    r2 = r2_score(y_test,y_pred)
    scores.append(r2)
    print(f'for alpha={i}, r2 score: {r2}')

print()
i_max = np.argmax(scores)
print("Best alpha : ",alphas[i_max])
print("Best score :", scores[i_max])

for alpha=0.25, r2 score: 0.6245764372818154
for alpha=0.5, r2 score: 0.6244646032856002
for alpha=0.75, r2 score: 0.6243503431206507
for alpha=1.0, r2 score: 0.6242338599592155
for alpha=1.25, r2 score: 0.6241153366520427
for alpha=1.5, r2 score: 0.6239949379104519
for alpha=1.75, r2 score: 0.6238728122366688
for alpha=2.0, r2 score: 0.6237490936338193
for alpha=2.25, r2 score: 0.6236239031227466
for alpha=2.5, r2 score: 0.6234973500892027
for alpha=2.75, r2 score: 0.6233695334818521
for alpha=3.0, r2 score: 0.623240542878878
for alpha=3.25, r2 score: 0.6231104594386885
for alpha=3.5, r2 score: 0.6229793567482498
for alpha=3.75, r2 score: 0.6228473015808715
for alpha=4.0, r2 score: 0.6227143545738014
for alpha=4.25, r2 score: 0.6225805708347099
for alpha=4.5, r2 score: 0.622446000485037
for alpha=4.75, r2 score: 0.6223106891472239
for alpha=5.0, r2 score: 0.6221746783819945
for alpha=5.25, r2 score: 0.622038006081148
for alpha=5.5, r2 score: 0.6219007068206661
for alpha=5.75, r2 score

In [59]:
alphas = np.arange(0,10,0.25)
scores = []
for i in alphas:
    ridge = Ridge(alpha=i)
    ridge.fit(x_train, y_train)
    y_pred = ridge.predict(x_test)
    r2 = r2_score(y_test,y_pred)
    scores.append(r2)
    print(f'for alpha={i}, r2 score: {r2}')

print()
i_max = np.argmax(scores)
print("Best alpha : ",alphas[i_max])
print("Best score : ", scores[i_max])

for alpha=0.0, r2 score: 0.6246856191453722
for alpha=0.25, r2 score: 0.6245764372818154
for alpha=0.5, r2 score: 0.6244646032856002
for alpha=0.75, r2 score: 0.6243503431206507
for alpha=1.0, r2 score: 0.6242338599592155
for alpha=1.25, r2 score: 0.6241153366520427
for alpha=1.5, r2 score: 0.6239949379104519
for alpha=1.75, r2 score: 0.6238728122366688
for alpha=2.0, r2 score: 0.6237490936338193
for alpha=2.25, r2 score: 0.6236239031227466
for alpha=2.5, r2 score: 0.6234973500892027
for alpha=2.75, r2 score: 0.6233695334818521
for alpha=3.0, r2 score: 0.623240542878878
for alpha=3.25, r2 score: 0.6231104594386885
for alpha=3.5, r2 score: 0.6229793567482498
for alpha=3.75, r2 score: 0.6228473015808715
for alpha=4.0, r2 score: 0.6227143545738014
for alpha=4.25, r2 score: 0.6225805708347099
for alpha=4.5, r2 score: 0.622446000485037
for alpha=4.75, r2 score: 0.6223106891472239
for alpha=5.0, r2 score: 0.6221746783819945
for alpha=5.25, r2 score: 0.622038006081148
for alpha=5.5, r2 score: